build/build.go

```diff
@@ -736,6 +737,13 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opt map[s
 			so, release, err := toSolveOpt(ctx, node, multiDriver, opt, np.bopts, configDir, w, func(name string) (io.WriteCloser, func(), error) {
 				return docker.LoadImage(ctx, name, w)
 			})
+			if desktop.BuildBackendEnabled() {
+				buildDetailsOpts[k] = desktop.BuildDetailsOpt{
+					Builder: node.Builder,
+					Node:    node.Name,
+					Ref:     so.Ref,
```

Comment:
We should only print if `Build()` ran and either was successful or we got error from BuildKit. Otherwise BuildKit has not registered the ref yet and it is useless.

Some options:
- Add link details based on status stream. If we get something on status stream then we know that buildkit has registered ref. This is also the time where we would write the link to progress in the future.
- In the `client.Build` error handler wrap the error with a custom error type. Eg. `RemoteBuildError` that does not change the `Error()` method but then caller can check which component returned the error with `errors.Is`.